pkg/loadbalancer/reconcile/deployment.go

```diff
@@ -0,0 +1,141 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconcile
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/open-telemetry/opentelemetry-operator/pkg/loadbalancer"
+)
+
+// +kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+
+// Deployments reconciles the deployment(s) required for the instance in the current context.
+func Deployments(ctx context.Context, params Params) error {
+	desired := []appsv1.Deployment{}
+	_, err := checkConfig(params)
+	if err != nil {
+		return fmt.Errorf("failed to parse Promtheus config: %v", err)
+	}
```

Comment:
Is this necessary?  This resource can be created without checking the config, though it may run into issues when  a pod tries to start and finds the config map volume missing.  If you think it is necessary, I'd suggest moving it inside the block guarded by `checkMode()`.  The same applies to the service reconciler.